docs/oauth.md

```diff
@@ -0,0 +1,169 @@
+* **Description** - A description of your application e.g. "Buckley's Bees sells the best honey in Ontario".
+* **Image** - An image that represents your application.
+* **Homepage URL** - The full URL to your application homepage e.g. "https://buckleysbees.ca".
+* **Redirect URL** - Your application's OAuth2 callback URL e.g. "https://buckleysbees.ca/oauth2/callback".
+
+Once you've registered your application you'll recieve your application credentials:
+
+* **Client ID** - A string identifying your application. This value is not secret and can be shown publicly e.g. `wm_id_BTQNPABtUqqApaDrcsDa`.
+* **Client Secret** - A string. This value is secret and must not be shared with anyone e.g. `wm_secret_iHmAWhvCwBYnuE6aZHcArBEPohanKmmWr8LcyBBPZYRkAxdmui`.
+
+You can register your application by emailing [jbuck](mailto:jon@mozillafoundation.org). Self-serve application registration is coming shortly.
+
+## Web Application Flow
+
+The web application flow is ideal for services that have a server-side component to securely store the client secret and user's access tokens.
+
+### 1. Redirect users to request access
+
+To start the web application flow you need to redirect your users browser to Webmaker.
+
+`GET https://id.webmaker.org/login/oauth/authorize`
+
+**Parameters**
+
+<table>
+  <tr>
+  	<th>Name</th>
+  	<th>Description</th>
+  </tr>
+  <tr>
+  	<td>client_id</td>
+  	<td>The client ID you received when you registered your application.</td>
+  </tr>
+  <tr>
+  	<td>response_type</td>
+  	<td>The response_type must be set to "code".</td>
+  </tr>
+  <tr>
+  	<td>scopes</td>
+  	<td>A space separated list of scopes.</td>
+  </tr>
+  <tr>
+  	<td>state</td>
+  	<td>An unguessable random string. It is used to protect against cross-site request forgery attacks.</td>
+  </tr>
+</table>
+
+**Example**
+
+`GET https://id.webmaker.org/login/oauth/authorize?client_id=wm_id_BTQNPABtUqqApaDrcsDa&response_type=code&state=Nvqfc67z`
+
+### 2. Webmaker redirects back to your application
+
+If the user accepts your request, Webmaker redirects back to your application with a temporary authorization code in a `code` parameter, your application's client ID in a `client_id` parameter, and the state you provided in the previous step in a `state` parameter.
+
+Your application must verify that the `state` and `client_id` parameters match the ones it provided when the sign-in process started. If you do not verify these parameters, a malicious party could hijack the sign-in process. 
+
+`GET https://buckleysbees.ca/oauth2/callback`
+
+**Parameters**
+
+<table>
+  <tr>
+  	<th>Name</th>
+  	<th>Description</th>
+  </tr>
+  <tr>
+  	<td>client_id</td>
+  	<td>The client ID you received when you registered your application.</td>
+  </tr>
+  <tr>
+  	<td>code</td>
+  	<td>A temporary authorization code</td>
+  </tr>
+  <tr>
+  	<td>state</td>
+  	<td>An unguessable random string. It is used to protect against cross-site request forgery attacks.</td>
```

Comment:
This is no longer "an", this is now "the random state value that you signed the original request with. If this value does not match the value sent with your initial request, callback processing should be cut off." or something to that effect